### DISCUSSION TITLE
Fails to install python@3.12 on codemagic-ci

### DISCUSSION THREAD

**Discussion by MehdiChinoune** [codemagic-ci-cd/codemagic-docs#2421] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Other - please mention in your question

### Steps to reproduce

This is about codemagic-ci
When It tries to install Python@3.12 with brew It fails to install and ask to install "Apple Command Line Tools"

### Expected results

Install successfully

### Actual results

Fails to install
```
Error: python@3.12: the bottle needs the Apple Command Line Tools to be installed.
  You can install them, if desired, with:
    xcode-select --install

If you're feeling brave, you can try to install from source with:
  brew install --build-from-source python@3.12

It is expected behaviour that most formulae will fail to build from source.
It is expected behaviour that Homebrew will be buggy and slow when building from source.
Do not create any issues about failures building from source on Homebrew's GitHub repositories.
Do not create any issues building from source even if you think this message is unrelated.
Any opened issues will be immediately closed without response.
Do not ask for help from Homebrew or its maintainers on social media.
You may ask for help building from source in Homebrew's discussions but are unlikely to receive a response.
If building from source fails, try to figure out the problem yourself and submit a fix as a pull request.
We will review it but may or may not accept it.
```

### Build id (optional)

65fa7c080f36be33daf5b6e4

**Comment by dtrdic** [comment, marked as answer]
Hello @MehdiChinoune, depending on the Xcode version that you specify in yaml file, Codemagic will use a different build machine type with different versions of preinstalled software. Please specify `xcode:15.3` under the environment section of your yaml, as this version comes with the Apple Command Line Tools preinstalled.

```
workflows:
 android-workflow:
    name: Android Workflow
    max_build_duration: 120
    instance_type: mac_mini_m1
    environment:
      xcode: 15.3
```

**Reply by MehdiChinoune** [reply]
I am using `xcode: latest`, so I should wait for https://github.com/orgs/codemagic-ci-cd/discussions/2407

**Reply by dtrdic** [reply]
This announcement only explains that if you use `xcode: latest`, the default version will be Xcode 15.3 starting from March 24. 
Xcode 15.3 already exists on Codemagic, and you can use it if you hardcode the version like i suggested above
